pulsar-broker-common/src/main/java/org/apache/pulsar/broker/authorization/PulsarAuthorizationProvider.java

```diff
@@ -283,59 +282,45 @@ public void initialize(ServiceConfiguration conf, PulsarResources pulsarResource
         }
 
         String topicUri = topicName.toString();
-        CompletableFuture<Void> future = pulsarResources.getNamespaceResources()
+        return pulsarResources.getNamespaceResources()
                 .setPoliciesAsync(topicName.getNamespaceObject(), policies -> {
                     policies.auth_policies.getTopicAuthentication()
                             .computeIfAbsent(topicUri, __ -> new HashMap<>())
                             .put(role, actions);
                     return policies;
-                }).thenRun(() -> {
-                    log.info("[{}] Successfully granted access for role {}: {} - topic {}", role, role, actions,
-                            topicUri);
+                }).whenComplete((__, throwable) -> {
+                    if (throwable != null) {
+                        log.error("[{}] Failed to set permissions for role {} on topic {}", role, role, topicName,
+                                throwable.getCause());
```

Comment:
```suggestion
                        log.error("[{}] Failed to set permissions for role {} on topic {}", role, role, topicName,
                                throwable);
```

Reply:
@mattisonchao - can you provide any insight here? I added the `.getCause()` because of your comment and some quick google searching. I thought the `.getCause()` was right because the `setPoliciesAsync` method has a `.thenApply` call. I'm not sure what the right answer is, though.

Reply:
I think both are ok here,  because we are only printing logs.

Reply:
Hi @michaeljmarshall 

Sorry to confuse you.
I think It's ok to use ``getCause`` or not when logging.

If we use it, we will get the stack trace like :

```java
java.util.concurrent.CompletionException: java.lang.IllegalArgumentException
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:314)
	at java.base/java.util.concurrent.CompletableFuture.uniApplyNow(CompletableFuture.java:683)
	at java.base/java.util.concurrent.CompletableFuture.uniApplyStage(CompletableFuture.java:658)
	at java.base/java.util.concurrent.CompletableFuture.thenApply(CompletableFuture.java:2094)
	at Scratch.main(scratch_3.java:6)
Caused by: java.lang.IllegalArgumentException
```

Without ``getCause`` : 

```java
java.lang.IllegalArgumentException
	at Scratch.lambda$main$0(scratch_3.java:7)
	at java.base/java.util.concurrent.CompletableFuture.uniApplyNow(CompletableFuture.java:680)
	at java.base/java.util.concurrent.CompletableFuture.uniApplyStage(CompletableFuture.java:658)
	at java.base/java.util.concurrent.CompletableFuture.thenApply(CompletableFuture.java:2094)
	at Scratch.main(scratch_3.java:6)
```

Reply:
No worries, and thanks for the explanation. Your comment showed me that futures do wrap failures in some cases, which I didn't know. I am going to remove the `.getCause()` calls, as @codelipenghui proposes.